### PR TITLE
Refactor `filling.repeat` API

### PR DESCRIPTION
**Pull request by isVoid**
Part of #10153 

This PR refactors `filling.repeat` cython API to accept a list of columns instead of Frame object. In this PR I'm also trying out a possibly better pattern for index and indexed_frame to share logics, which might become a solution for https://github.com/rapidsai/cudf/issues/10068.

### CODE REVIEW

File: python/cudf/cudf/core/indexed_frame.py [hunk 1]
```diff
@@ -1778,6 +1778,79 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
 
         return NotImplemented
 
+    @annotate("FRAME_REPEAT", color="green", domain="cudf_python")
+    def repeat(self, repeats, axis=None):
+        """Repeats elements consecutively.
+
+        Returns a new object of caller type(DataFrame/Series/Index) where each
```
Comment (vyasr) [inline python/cudf/cudf/core/indexed_frame.py:1785]:
```suggestion
        Returns a new object of caller type(DataFrame/Series) where each
```

File: python/cudf/cudf/core/_base_index.py [hunk 2]
```diff
@@ -1518,6 +1518,37 @@ def _apply_boolean_mask(self, boolean_mask):
...
+        --------
+        >>> index = cudf.Index([10, 22, 33, 55])
+        >>> index
+        Int64Index([10, 22, 33, 55], dtype='int64')
+        >>> index.repeat(5)
+        Int64Index([10, 10, 10, 10, 10, 22, 22, 22, 22, 22, 33,
+                    33, 33, 33, 33, 55, 55, 55, 55, 55],
+                dtype='int64')
+        """
+        raise NotImplementedError()
```
Comment (vyasr) [inline python/cudf/cudf/core/_base_index.py:1550]:
```suggestion
        raise NotImplementedError
```

**Review by vyasr — APPROVED** [review]
LGTM!

**Comment by vyasr** [comment]
rerun tests

**Comment by isVoid** [comment]
rerun tests

**Comment by isVoid** [comment]
rerun tests

**Comment by isVoid** [comment]
rerun tests

**Pull request merged**